src/main/scala/mesosphere/marathon/state/AppDefinition.scala

```diff
@@ -769,6 +770,13 @@ object AppDefinition extends GeneralPurposeCombinators {
     }
   }
 
+  private val complyWithMesosHealthChecks: Validator[AppDefinition] =
+    conditional[AppDefinition](appDef =>
+      (appDef.healthChecks.count(_.isInstanceOf[MesosHealthCheck]) -
+        appDef.healthChecks.count(_.isInstanceOf[MesosCommandHealthCheck])) > 0) {
+      featureEnabled(Features.MESOS_HEALTHCHECKS)
```

Comment:
tl;dr: fine with me.

Long version:
We don't check master version for other feature flags, so I can't say we must do that here. Yet in general, it would be better to combine feature flags with a minimum master version to make sure that the user gets a meaningful error message when trying to enable a feature that isn't supported by the used mesos version.

Although Mesos tries to be pretty much backwards compatible, I'm sure I have seen invalid combinations of super native library with a newer Mesos. Current master code checks for both libmesos version and master version [here](https://github.com/mesosphere/marathon/blob/23cb098275f0db1a9afda0d59ca9144e99536937/src/main/scala/mesosphere/mesos/LibMesos.scala#L7)

The `MesosMasterMinimumVersion` is checked against after having registered – if the version reported by the frameworkInfo is less, Marathon will throw and commit suicide. Libmesos will be checked during startup (even before registering).

Btw the version as reported by libmesos returned bogus values until [MESOS-6502](https://issues.apache.org/jira/browse/MESOS-6502) was fixed ([here](https://github.com/apache/mesos/commit/cbbc5a46a499eb125eac83af1f21b832f136385a))